include/ozo/connection_info.h

```diff
@@ -11,29 +12,28 @@ template <
     typename OidMap = empty_oid_map,
     typename Statistics = no_statistics>
 class connection_info {
-    io_context& io_;
-    std::string conn_str_;
-    Statistics statistics_;
-
-    using connection = impl::connection_impl<OidMap, Statistics>;
+    std::string conn_str;
+    Statistics statistics;
 
 public:
+    using connection = impl::connection_impl<OidMap, Statistics>;
     using connection_type = std::shared_ptr<connection>;
+    using source_type = connection_info;
```

Comment:
IMO **source_type** is suitable only for the **connector** class because in other cases this type points to different entities what is misleading. From this PR I can see no application for this type outside of **connector** class. If you have any ideas about the future of this type, please describe it in few words.